armadillo/src/main/java/at/favre/lib/armadillo/SecureSharedPreferences.java

```diff
@@ -253,6 +256,45 @@ public void unregisterOnSharedPreferenceChangeListener(OnSharedPreferenceChangeL
         sharedPreferences.unregisterOnSharedPreferenceChangeListener(onSharedPreferenceChangeListener);
     }
 
+    private final List<SharedPreferenceChangeListenerWrapper> securePreferenceListeners = new LinkedList<>();
```

Comment:
Why do we need an additional list managing the `securePreferenceListeners`? Wouldn't it suffice to just wrap it in `new SharedPreferenceChangeListenerWrapper();` and add it to the internal listener map? This would remove a lot of the complexity of this code.

Reply:
What I wanted to achieve is to allow the client code to register/unregister using the original `OnSecurePreferenceChangeListener` and if I just wrapped this one with the `SharedPreferenceChangeListenerWrapper` and pass it down to the _SharedPreference's own listener management_ I didn't know how to do the **unregister** part. Do you have an idea for doing it? I think I might be missing something here.

Reply:
Fair enough, wrapping it would make it another reference the user does not know, so the unregister wouldn't work. I'd prefer the cleaner shared preference managed version, but do not have a better suggestions.